### PR TITLE
Update Transformer App

### PR DESCRIPTION
**Pull request by fiedorowicz1**
Fix several issues with the transformer app and generally update it by taking advantage of Hugging Face datasets and metrics.

### CODE REVIEW

File: applications/nlp/transformer/evaluate_model.py [hunk 1]
```diff
@@ -272,37 +284,38 @@ def evaluate_transformer(weights_prefix):
...
-    classifier = torch.nn.Linear(embed_dim, vocab_size, bias=False)
+    embedding_layer = load_embedding_layer(weights_prefix).to(device)
+    transformer = load_transformer(weights_prefix).to(device)
+    classifier = torch.nn.Linear(embed_dim, vocab_size, bias=False).to(device)
     classifier.weight = embedding_layer.weight
 
     # Evaluate model
     bleu_scores = []
-    for batch, index_start in enumerate(range(0, num_samples, mini_batch_size)):
+    for batch, index_start in enumerate(tqdm(range(0, num_samples, mini_batch_size))):
```
Comment (tbennun) [inline applications/nlp/transformer/evaluate_model.py:294]: This adds `tqdm` as a requirement, which it wasn't before.

**Review by tbennun — CHANGES_REQUESTED** [review]

File: applications/nlp/transformer/dataset.py [hunk 2]
```diff
@@ -1,9 +1,10 @@
-"""WMT 2014 dataset for English-German translation."""
+"""WMT 2016 dataset for English-German translation."""
 import os.path
 import sys
 
 import numpy as np
-import torchnlp.datasets
+from datasets import load_dataset
+from tokenizers import Tokenizer
```
[the remark this answers is not in the document]
Comment (fiedorowicz1) [reply applications/nlp/transformer/dataset.py:8]: Updated the readme to include dependency information.

File: applications/nlp/transformer/evaluate_model.py (hunk 1 above)
Comment (fiedorowicz1) [reply applications/nlp/transformer/evaluate_model.py]: Removed

File: applications/nlp/utils/paths.py [hunk 3]
```diff
@@ -18,8 +18,19 @@ def root_dir():
...
+    See https://huggingface.co/datasets/wmt16
+
+    The dataset has already been downloaded on LLNL LC systems and is
+    available to anyone in the "lbann" group.
+
+    """
+
+    return '/p/vast1/lbann/datasets/wmt16_en_de_huggingface'
+
+def wmt_dir_old(system=system()):
```
[the remark this answers is not in the document]
Comment (fiedorowicz1) [reply applications/nlp/utils/paths.py]: Removed

**Review by tbennun — APPROVED** [review]

**Review by ndryden — APPROVED** [review]
Looks good to me. Would be nice to support the new protobin stuff for weights, but that's probably a separate PR.
